aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -224,13 +229,23 @@ public BaseHandlerStd(final HandlerConfig config) {
             final ResourceHandlerRequest<ResourceModel> request,
             final CallbackContext context,
             final Logger logger) {
-        return handleRequest(
-                proxy,
-                request,
-                context != null ? context : new CallbackContext(),
-                proxy.newProxy(RdsClientBuilder::getClient),
-                proxy.newProxy(Ec2ClientBuilder::getClient),
-                logger);
+        LoggerWrapper loggerWrapper = new LoggerWrapper(logger, request);
+        logRequest(loggerWrapper, request);
+        ProgressEvent<ResourceModel, CallbackContext> progressEvent = null;
+        try {
+            progressEvent = handleRequest(
+                    proxy,
+                    request,
+                    context != null ? context : new CallbackContext(),
+                    proxy.newProxy(RdsClientBuilder::getClient),
+                    proxy.newProxy(Ec2ClientBuilder::getClient),
+                    logger
+            );
+            logResponse(loggerWrapper, progressEvent);
+        } catch (Throwable throwable) {
+            loggerWrapper.logThrowableAndRethrow(throwable);
+        }
+        return progressEvent;
```

Comment:
Looks a bit too massive. What if we turn it into a proxy-function instead? It also could be an annotation, which makes it even more elegant.